### PR TITLE
DEV: Add markdown_additional_options to Site

### PR DESCRIPTION
**Pull request by martin-brennan**
Sometimes plugins need to have additional data or options available
when rendering custom markdown features/rules that are not available
on the default opts.discourse object. These additional options should
be namespaced to the plugin adding them.

```
Site.markdown_additional_options["chat"] = { limited_pretty_text_markdown_rules: [] }
```

These are passed down to markdown rules on opts.discourse.additionalOptions.

The main motivation for adding this is the chat plugin, which currently stores
chat_pretty_text_features and chat_pretty_text_markdown_rules on
the Site object via additions to the serializer, and the Site object is
not accessible to import via markdown rules (either through
Site.current() or through container.lookup). So, to have this working
for both front + backend code, we need to attach these additional options
from the Site object onto the markdown options object.

### CODE REVIEW

File: app/models/site.rb [hunk 1]
```diff
@@ -7,6 +7,20 @@ class Site
   cattr_accessor :preloaded_category_custom_fields
   self.preloaded_category_custom_fields = Set.new
 
+  ##
```
Comment (CvX) [inline app/models/site.rb:10]: btw, what's this style of `##` comments? 🤔

**Review by CvX — APPROVED** [review]

File: app/assets/javascripts/discourse/app/lib/text.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ function getOpts(opts) {
       formatUsername,
       watchedWordsReplace: context.site.watched_words_replace,
       watchedWordsLink: context.site.watched_words_link,
+      additionalOptions: context.site.markdown_additional_options,
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/lib/text.js:26]: Are we able to just collapse `context.site.censored_regexp`, `context.site.custom_emoji_translation` etc into a single `context.site.markdown_options` object? It'll provide a cleaner interface for all the engines to consume additional options that they need from the site's context.

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/lib/text.js:26]: @tgxworld I'd rather not move stuff around in the same PR, I just need to be unblocked on this chat stuff. I will do a follow up PR to move things around.

File: app/models/site.rb (hunk 1 above)
Comment (martin-brennan) [reply app/models/site.rb:10]: @CvX IDK where exactly I picked that up from, maybe yard? I usually do them to kind of "fence in" the top of big blocks of comments

**Pull request merged**